packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx

```diff
@@ -46,21 +70,24 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
         root: {
             backgroundColor: "white",
         },
-        columnHeader: {
+        columnHeader: ({ ownerState }) => ({
             "&:focus": {
                 outline: "none",
             },
             "&:focus-within": {
                 outline: "none",
             },
-        },
+            ...getDensityStyles(ownerState?.density, spacing),
+        }),
         pinnedColumns: {
             backgroundColor: "white",
             boxShadow: shadows[2],
         },
+        row: ({ ownerState }) => ({
+            ...getDensityStyles(ownerState?.density, spacing),
```

Comment:
Setting the height only on `row` causes `cell` to be misaligned. 
The height probably needs to be either moved to `cell` or add to `cell` in addition to `row`. 
 
<img width="925" alt="Row misaligned" src="https://github.com/user-attachments/assets/0946fd04-13bd-4da6-8610-a39752317099" />

The same seems to be the case for the footer:
<img width="923" alt="Footer misaligned" src="https://github.com/user-attachments/assets/a82b388d-3eb6-4b3f-8411-7ef92435c7ee" />